src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -503,6 +503,43 @@ internal static IEnumerable<string> GetFrameworkAssembliesToTreatAsUserAssemblie
 			return acw_map;
 		}
 
+		public static Dictionary<string, HashSet<string>> LoadMapFile (IBuildEngine4 engine, string mapFile)
+		{
+			var cachedMap = (Dictionary<string, HashSet<string>>)engine?.GetRegisteredTaskObject (mapFile, RegisteredTaskObjectLifetime.Build);
+			if (cachedMap != null)
+				return cachedMap;
+			var map = new Dictionary<string, HashSet<string>> ();
+			if (!File.Exists (mapFile))
+				return map;
+			foreach (var s in File.ReadLines (mapFile)) {
+				var items = s.Split (';');
+				var key = items [0];
+				var value = items [1];
+				if (!map.ContainsKey (key))
+					map.Add (key, new HashSet<string> ());
+				if (!map[key].Contains (value))
```

Comment:
You don't need `.Contains()` here, as it's a `HashSet<T>`.  Just always call `map [key].Add(value)`.